assignment01-voronoi/worldgen-voronoi.c

```diff
@@ -0,0 +1,141 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <time.h>
+#include <stdlib.h>
+#include "tiles.h"
+#include <math.h>
+
+
+#define GRID_HEIGHT 24-3
+#define GRID_WIDTH 80
+#define SEED_NUM 20
+
+static char gridMatrix[GRID_HEIGHT][GRID_WIDTH];
+
+typedef struct Seed{
+    int x;
+    int y;
+    char tile;
+} Seed;
+
+
+void init_world() 
+{
+   for(int i=0; i < GRID_HEIGHT; i++) {
+       	gridMatrix[i][0] = mountainTile.ascii;
+	gridMatrix[i][GRID_WIDTH-1] =  mountainTile.ascii;
+   }
+
+   for(int j=0; j < GRID_WIDTH; j++) {
+       gridMatrix[0][j] = mountainTile.ascii;
+       gridMatrix[GRID_HEIGHT-1][j] =  mountainTile.ascii;
+   }
+
+   for(int i=1; i < GRID_HEIGHT - 1; i++) {
+       for(int j=1; j < GRID_WIDTH - 1; j++) {
+	   gridMatrix[i][j] =  clearing.ascii;
+       }
+   }
+}
+
+void printGrid()
+{
+    //	printf("\x1b[H");
+    for(int i=0; i < GRID_HEIGHT; i++) {
+	for(int j=0; j < GRID_WIDTH; j++) {
+	  // 	putchar(gridMatrix[i][j]);
+	    if(gridMatrix[i][j] == '%'){
+		printf(MAG "%c", gridMatrix[i][j]);
```

Comment:
im not even sure how this works mag and "%c" are treated as 2 separate strings

Reply:
the miracle of ANSII escape codes, i dont even know how they work.